internal/report/store.go

```diff
@@ -1,291 +1,45 @@
-	if getErr != nil {
-		if errorMachinery.IsNotFound(getErr) {
-			return ClusterPolicyReport{}, constants.ErrResourceNotFound
-		}
-		return ClusterPolicyReport{}, getErr
-	}
-	return ClusterPolicyReport{
-		result,
-	}, nil
-}
+var SupportedTypes = [2]string{KUBERNETES, MEMORY}
 
-// Update namespaced PolicyReport
-func (s *PolicyReportStore) UpdatePolicyReport(report *PolicyReport) error {
-	err := s.client.Update(context.Background(), &report.PolicyReport)
-	if err != nil {
-		return err
-	}
-	summary, _ := report.GetSummaryJSON()
-	log.Info().
-		Dict("dict", zerolog.Dict().
-			Str("report name", report.GetName()).
-			Str("report ns", report.GetNamespace()).
-			Str("report resourceVersion", report.GetResourceVersion()).
-			Str("summary", summary),
-		).Msg("updated PolicyReport")
-	return nil
-}
+// PolicyReportStore caches the latest version of PolicyReports
+// There is 2 different stores available:
+//   - KubernetesPolicyReportStore, which uses K8s/etcd backend only
+//   - MemoryPolicyReportStore, which uses in-memory cache
+type PolicyReportStore interface {
+	// GetPolicyReport returns the Policy Report defined inside a given namespace.
+	// An empty PolicyReport is returned when nothing is found
+	GetPolicyReport(namespace string) (PolicyReport, error)
 
-// Update ClusterPolicyReport or PolicyReport. ns argument is used in case
-// of namespaced PolicyReport
-func (s *PolicyReportStore) UpdateClusterPolicyReport(report *ClusterPolicyReport) error {
-	err := s.client.Update(context.Background(), &report.ClusterPolicyReport)
-	if err != nil {
-		return err
-	}
-	summary, _ := report.GetSummaryJSON()
-	log.Info().
-		Dict("dict", zerolog.Dict().
-			Str("report name", report.GetName()).
-			Str("report ns", report.GetNamespace()).
-			Str("summary", summary),
-		).Msg("updated ClusterPolicyReport")
-	return nil
-}
+	// GetClusterPolicyReport gets the ClusterPolicyReport
+	GetClusterPolicyReport(name string) (ClusterPolicyReport, error)
 
-// Delete PolicyReport by namespace
-func (s *PolicyReportStore) RemovePolicyReport(namespace string) error {
-	if report, err := s.GetPolicyReport(namespace); err == nil {
-		err := s.client.Delete(context.Background(), &report.PolicyReport)
-		if err != nil {
-			return err
-		}
-	}
-	return nil
-}
+	// UpdatePolicyReport updates namespaced PolicyReport
+	UpdatePolicyReport(report *PolicyReport) error
 
-// Delete all namespaced PolicyReports
-func (s *PolicyReportStore) RemoveAllNamespacedPolicyReports() error {
-	err := s.client.DeleteAllOf(context.Background(), &polReport.PolicyReport{},
-		client.MatchingLabels(map[string]string{
-			LabelAppManagedBy: LabelApp,
-		}))
-	if err != nil {
-		return err
-	}
-	return nil
-}
+	// UpdateClusterPolicyReport updates a ClusterPolicyReport or PolicyReport.
+	// ns argument is used in case of namespaced PolicyReport
+	UpdateClusterPolicyReport(report *ClusterPolicyReport) error
```

Comment:
Since only `SaveClusterPolicyReport` and `SavePolicyReport` are used from outside of the package, we should not have `UpdateClusterPolicyReport` and `UpdatePolicyReport` as part of the interface, and they should be private on the implemented packages.

Reply:
to be honest, I had mixed feelings about this one, I did notice that these 2 methods were not being used outside of the package but I didn't want to change the interface since you could've left them public for future work. in this case, I'll remove them from the interface

Reply:
also, with that in mind, does it make sense to remove both `RemovePolicyReport` and `RemoveAllNamespacedPolicyReports` given that we currently have no usage for them?

Reply:
Yeah, if we are not using, I'm fine removing these methods. I think that  code no in use should not be in the code base.

Reply:
completely agree @jvanz , I'll remove those `Remove*` methods